go/pkg/client/cas.go

```diff
@@ -722,53 +904,97 @@ func (c *Client) DownloadFiles(ctx context.Context, execRoot string, outputs map
 		}
 	}
 
-	eg, eCtx := errgroup.WithContext(ctx)
+	ctx := context.Background()
 	for i, batch := range batches {
-		i, batch := i, batch     // https://golang.org/doc/faq#closures_and_goroutines
-		c.casDownloaders <- true // Reserve an downloader thread.
-		eg.Go(func() error {
-			defer func() { <-c.casDownloaders }()
+		i, batch := i, batch // https://golang.org/doc/faq#closures_and_goroutines
+		go func() (err error) {
```

Comment:
err is ignored?

Reply:
No, not ignored -- it is handled by the deferred afterDownload in lone 935.

Reply:
Then better to declare err in the beginning of this function and deferred call afterDownload there?

`return err` in this function seems meaningless and bit confusing to me.

Reply:
This will not do the same thing -- note that I only defer afterDownload for the case of a single-blob batch; for the multi-blob batch case I handle everything manually. Perhaps, to make this clearer, I could split the function into two functions `readBatch` and `readSingle`, and only return/defer on the readSingle one? I'm not sure it will be clearer, though...

Reply:
I'm also not clear on how the goroutine's returned err is handled. If you're using them then maybe use an errgroup?